src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -338,27 +339,78 @@ export function resolveConfiguredUntrustedSettings(groups: ISettingsGroup[], tar
 	return [...allSettings].filter(setting => setting.restricted && inspectSetting(setting.key, target, configurationService).isConfigured);
 }
 
-export function resolveExtensionsSettings(groups: ISettingsGroup[]): ITOCEntry<ISetting> {
-	const settingsGroupToEntry = (group: ISettingsGroup) => {
+export async function resolveExtensionsSettings(extensionService: IExtensionService, groups: ISettingsGroup[]): Promise<ITOCEntry<ISetting>> {
+	const extGroupTree = new Map<string, ITOCEntry<ISetting>>();
+	const addEntryToTree = (extensionId: string, extensionName: string, childEntry: ITOCEntry<ISetting>) => {
+		if (!extGroupTree.has(extensionId)) {
+			const rootEntry = {
+				id: extensionId,
+				label: extensionName,
+				children: []
+			};
+			extGroupTree.set(extensionId, rootEntry);
+		}
+		extGroupTree.get(extensionId)!.children!.push(childEntry);
+	};
+	const processGroupEntry = async (group: ISettingsGroup) => {
 		const flatSettings = arrays.flatten(
 			group.sections.map(section => section.settings));
 
-		return {
+		const extensionId = group.extensionInfo!.id;
+		const extension = await extensionService.getExtension(extensionId);
+		const extensionName = extension!.displayName ?? extension!.name;
+
+		let sectionOrder = 0;
+		const extensionConfiguration = extension?.contributes?.configuration;
+		if (Array.isArray(extensionConfiguration)) {
+			const currentSection = extensionConfiguration.find((config) => (config as any)?.id === group.id);
```

Comment:
The `id` and `order` fields are not stored in `IConfiguration`, so this code retrieves them indirectly.
It probably wouldn't hurt to add them as optional fields into the interface, assuming this use case is fine.

Reply:
Yeah,  add it to whatever interface it needs to be on instead of looking it up this way